src/nspawn/nspawn-patch-uid.c

```
@@ -239,10 +240,14 @@ static int patch_fd(int fd, const char *name, const struct stat *st, uid_t shift
 
                 /* The Linux kernel alters the mode in some cases of chown(). Let's undo this. */
                 if (name) {
+                        /* It looks like older glibc (before 2016) did not support AT_SYMLINK_NOFOLLOW. */
                         if (!S_ISLNK(st->st_mode))
-                                r = fchmodat(fd, name, st->st_mode, 0);
-                        else /* AT_SYMLINK_NOFOLLOW is not available for fchmodat() */
-                                r = 0;
+                                r = fchmodat_best(fd, name, st->st_mode, 0);
+                        else {
+                                r = fchmodat2(fd, name, st->st_mode, AT_SYMLINK_NOFOLLOW);
```

Please just cast this to void if error is ignored.

---

(Or handle the error other than ENOSYS here)